frontend/src/container/GridCardLayout/GridCardLayout.tsx

```diff
@@ -0,0 +1,132 @@
+import { PlusOutlined, SaveFilled } from '@ant-design/icons';
+import { SOMETHING_WENT_WRONG } from 'constants/api';
+import { PANEL_TYPES } from 'constants/queryBuilder';
+import { useUpdateDashboard } from 'hooks/dashboard/useUpdateDashboard';
+import useComponentPermission from 'hooks/useComponentPermission';
+import { useIsDarkMode } from 'hooks/useDarkMode';
+import { useNotifications } from 'hooks/useNotifications';
+import { useDashboard } from 'providers/Dashboard/Dashboard';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { AppState } from 'store/reducers';
+import { Dashboard, Widgets } from 'types/api/dashboard/getAll';
+import AppReducer from 'types/reducer/app';
+
+import { headerMenuList } from './config';
+import GridCard from './GridCard';
+import {
+	Button,
+	ButtonContainer,
+	Card,
+	CardContainer,
+	ReactGridLayout,
+} from './styles';
+import { GraphLayoutProps } from './types';
+
+function GraphLayout({
+	onAddPanelHandler,
+	widgets,
+}: GraphLayoutProps): JSX.Element {
+	const { selectedDashboard, layouts, setLayouts } = useDashboard();
+	const { t } = useTranslation(['dashboard']);
+
+	const { featureResponse } = useSelector<AppState, AppReducer>(
```

Comment:
Multiple/similar `useSelector`
```suggestion
	const { featureResponse, role } = useSelector<AppState, AppReducer>(
```